### PR TITLE
displays.add_overlay: Add check if ims is empty

### PR DESCRIPTION
**Pull request by alexsavio**
Hi,

This fix I think it makes sense since BaseSlices._map_show can actually return an empty list.

### CODE REVIEW

File: nilearn/plotting/displays.py [hunk 1]
```diff
@@ -584,7 +584,7 @@ def add_overlay(self, img, threshold=1e-6, colorbar=False, **kwargs):
         kwargs.setdefault('interpolation', 'nearest')
         ims = self._map_show(img, type='imshow', threshold=threshold, **kwargs)
 
-        if colorbar:
+        if colorbar and ims:
```
Comment (GaelVaroquaux) [inline nilearn/plotting/displays.py]: That is actually a surprising line for me. Could you add a comment explaining when ims doesn't evaluate to true (ie it's an empty list?)

**Comment by GaelVaroquaux** [comment]
Also, if you have a test case for the bug that this is fixing, it would be great.

**Comment by alexsavio** [comment]
In the issue #1135 I linked a small example where this happens.
In this example the files don't have exactly the same matrix shape, I was expecting a 'shape mismatch' error, but I guess you already added the resampling for plotting.

In my use case I want to plot a subset of the cuts that I get with find_cut_slices.
The thing is that I have it solved already with this PR.

Here goes the test:
```python
import nilearn.plotting as niplot
import nilearn.image as niimg

img    = niimg.load_img('test.nii.gz')  # shape is (79,  95, 79)
bg_img = 'single_subj_T1_brain.nii.gz'  # shape is (91, 109, 91)

cuts = niplot.find_cut_slices(img, 
                              n_cuts=20,
                              direction='z')

p = niplot.plot_stat_map(img, 
                         display_mode='z', 
                         cut_coords=cuts[-8:],)
                         bg_img=bg_img)

niplot.show()

p.close()
```

**Comment by GaelVaroquaux** [comment]
Yes, indeed. But we would need a test as in a unit test, for example on small artificial data, in https://github.com/nilearn/nilearn/blob/master/nilearn/plotting/tests/test_displays.py. This is important because it gets run by the continuous integration and prevents regression in the code.

File: nilearn/plotting/tests/test_img_plotting.py [hunk 2]
```diff
@@ -854,3 +856,28 @@ def test_invalid_in_display_mode_cut_coords_all_plots():
                             "be a list of 3d world coordinates.",
                             plot_func,
                             img, display_mode='ortho', cut_coords=2)
+
+
+def test_outlier_cut_coords():
+    """ Test to plot a few cuts that are out of the """
```
Comment (GaelVaroquaux) [inline nilearn/plotting/tests/test_img_plotting.py]: Looks like you forgot to end the sentence.

File: nilearn/plotting/displays.py (hunk 1 above)
Comment (GaelVaroquaux) [inline nilearn/plotting/displays.py]: Can you also address this comment please :)

I am thinking about people looking at the code in the future.

**Comment by alexsavio** [comment]
I didn't know exactly what to put there, since I still haven't gone through `_map_show`.

**Comment by alexsavio** [comment]
Not sure if this is enough. If you have patience, a few days, I can decipher _map_show and add comments and docstrings to it.

**Comment by GaelVaroquaux** [comment]
LGTM. Thank you very much.

Merging.

**Pull request merged**